apps/customquotes/custom_quotes.star

```diff
@@ -0,0 +1,153 @@
+
+        # Randomly pick one quote-author
+        quote_id = int(time.now().nanosecond / 1000) % len(quotes)
+        if (quote_id == 0):
+            quote_id = quote_id + 1
+        quote = quotes[quote_id]
+        author = authors[quote_id]
+
+        # Display error if no quote
+        if (quote == ""):
+            quote = "No quote to display"
+
+        # Add hyphen if there's an author
+        if (author != ""):
+            author = "-" + author
+
+    image = base64.decode(get_image())
+
+    return render.Root(
+        show_full_animation = True,
+        delay = 200,
+        child = render.Stack(
+            children = [
+                render.Image(src = image, height = OUTER_HEIGHT, width = OUTER_WIDTH),
+                render.Padding(
+                    pad = PADDING,
+                    child = render.Box(
+                        color = BOX_COLOR,
+                        height = INNER_HEIGHT,
+                        width = INNER_WIDTH,
+                        child = render.Marquee(
+                            height = INNER_HEIGHT,
+                            width = INNER_WIDTH,
+                            child = render.Column(
+                                main_align = "start",
+                                cross_align = "center",
+                                children = [
+                                    render.WrappedText(
+                                        content = quote,
+                                        font = QUOTE_FONT,
+                                        color = QUOTE_COLOR,
+                                        linespacing = 1,
+                                        width = INNER_WIDTH,
+                                    ),
+                                    render.Box(width = INNER_WIDTH, height = 1),
+                                    render.WrappedText(
+                                        content = author,
+                                        font = AUTHOR_FONT,
+                                        color = AUTHOR_COLOR,
+                                        linespacing = 2,
+                                        width = INNER_WIDTH,
+                                    ),
+                                ],
+                            ),
+                            scroll_direction = "vertical",
+                            offset_start = 16,
+                            align = "end",
+                        ),
+                    ),
+                ),
+            ],
+        ),
+    )
+
+# Define function to get random image
+def get_image():
+    image = cache.get("image")
+    if not image:
+        response = http.get("https://random.imagecdn.app/500/250")
+        if response.status_code != 200:
+            fail("Failed to retrieve image: %d - %s" % (response.status_code, response.body()))
+        image = base64.encode(response.body())
+
+        cache.set("image", image, ttl_seconds = TTL_SECONDS)
+    return image
```

Comment:
Please use `ttl_seconds` in `http.get` instead of using cache directly.

Example at: https://discuss.tidbyt.com/t/http-caching-client/5129

Reply:
Thanks. I made your suggested changes.